runtime/unix.c

```diff
@@ -426,6 +433,41 @@ char *caml_secure_getenv (char const *var)
 #endif
 }
 
+int64_t caml_time_counter(void)
+{
+#if defined(HAS_MACH_ABSOLUTE_TIME)
+  static mach_timebase_info_data_t time_base = {0};
+  uint64_t now;
+
+  if (time_base.denom == 0) {
+    if (mach_timebase_info(&time_base) != KERN_SUCCESS)
+      return 0;
+
+    if (time_base.denom == 0)
+      return 0;
```

Comment:
Is this second check necessary?  mach_timebase_info guarantees to fill in time_base.denom if it returns KERN_SUCCESS.

Reply:
I'm actually not sure, I've restored this from pre-merge multicore where it used to exist. Do you have a link for the documentation for mach_timebase_info? The Apple docs seem pretty useless.

Reply:
Thanks @sabine and @avsm . I think it's probably safe to remove the redundant check, I've done that in https://github.com/ocaml/ocaml/pull/10964/commits/0e1ced41326c9c9cf1a0ab25c0054e8be8d8cf94